Number_types/include/CGAL/Quotient.h

```diff
@@ -685,12 +785,239 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
 
     class To_interval
       : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
+
       public:
+
+        bool are_correct_bounds( const double l, const double u, const Type& x ) const {
+
+          if (
+            CGAL::abs(l) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(u) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(l) == 0.0 ||
+            CGAL::abs(u) == 0.0) {
+            return true;
+          }
+          CGAL_assertion(CGAL::abs(l) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(u) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(l) != 0.0);
+          CGAL_assertion(CGAL::abs(u) != 0.0);
+
+          const Type lb(l), ub(u);
+          CGAL_assertion(lb <= ub);
+          CGAL_assertion(lb <= x);
+          CGAL_assertion(ub >= x);
+          return lb <= ub && lb <= x && ub >= x;
```

Comment:
The first test is redundant with the other 2.

Reply:
Removed.